### ISSUE TITLE
Support configurable cache times for datasources

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#6573]
**What would you like Renovate to be able to do?**

Allow self-hosted users to customize the cache TTL for datasource results such as npm and maven.

**Describe the solution you'd like**

Needs to allow varying default values like we have today, but also be configurable.

**Additional context**

See #6572 for a related topic

**Comment by viceice** [comment]
Wouldn't this be easy possible with host rules? 🤔

**Comment by rarkins** [comment]
Yes if we think people want to configure them differently per host for the same datasource. But if you just want a common cache time per-datasource shared between all registries then packageRules are better suited.

**Comment by rarkins** [comment]
Yes if we think people want to configure them differently per host for the same datasource. But if you just want a common cache time per-datasource shared between all registries then packageRules are better suited.

**Comment by philippe-granet** [comment]
@rarkins current workaround I use to increase concurrent request to registry from 5 to 50 and cache TTL from 10mins to 24h :
```
sed -i 's/10/24 * 60/g' /usr/src/app/node_modules/renovate/dist/datasource/maven/index.js
sed -i 's/concurrency: 5/concurrency: 50/g' /usr/src/app/node_modules/renovate/dist/datasource/maven/index.js`
```
Support configurable concurrency request to registry could be implemented with this issue or may I open a new issue?

**Comment by rarkins** [comment]
@philippe-granet please follow and comment on #6606

**Comment by JoeJennings** [comment]
@rarkins ->  I suspect this would be a nice performance enhancement for some projects and allow for a reduction in request traffic to datasources.  Any thoughts on when if this might be implemented?
